### PR TITLE
Ekco 0.9.0

### PR DESCRIPTION
**Pull request by areed**

### CODE REVIEW

File: addons/ekco/0.9.0/install.sh [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    fi
+    EKCO_PURGE_DEAD_NODES=false
+    if [ "$EKCO_SHOULD_ENABLE_PURGE_NODES" = "1" ]; then
+        EKCO_PURGE_DEAD_NODES=true
+    fi
+    EKCO_CLEAR_DEAD_NODES=true
+    if [ "$EKCO_SHOULD_DISABLE_CLEAR_NODES" = "1" ]; then
+        EKCO_CLEAR_DEAD_NODES=false
+    fi
+    if [ "$ROOK_VERSION" = "1.0.4" ]; then
```
Comment (areed) [inline addons/ekco/0.9.0/install.sh:28]: This block is new

File: addons/ekco/0.9.0/install.sh [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    cp "$src/deployment.yaml" "$dst/deployment.yaml"
+    cp "$src/rotate-certs-rbac.yaml" "$dst/rotate-certs-rbac.yaml"
+
+    # is rook enabled
+    if kubectl get ns | grep -q rook-ceph; then
+        cp "$src/rbac-rook.yaml" "$dst/rbac-rook.yaml"
+        insert_resources "$dst/kustomization.yaml" rbac-rook.yaml
+        cat "$src/rolebinding-rook.yaml" >> "$dst/rolebinding.yaml"
+
+        if [ -n "$EKCO_ROOK_PRIORITY_CLASS" ]; then
```
Comment (areed) [inline addons/ekco/0.9.0/install.sh:50]: This block is new

File: addons/ekco/0.9.0/rbac-rook.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  - apiGroups: ["ceph.rook.io"]
+    resources:
+      - cephclusters
+      - cephblockpools
+      - cephfilesystems
+      - cephobjectstores
+    verbs:
+      - get
+      - update
+  - apiGroups: ["scheduling.k8s.io"]
```
Comment (areed) [inline addons/ekco/0.9.0/rbac-rook.yaml:32]: Rest of this file is new

File: addons/ekco/0.9.0/rbac.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: ekco-cluster
+rules:
+  - apiGroups: [""]
+    resources:
+      - nodes
+      - pods
+      - secrets
+      - services
```
Comment (areed) [inline addons/ekco/0.9.0/rbac.yaml:17]: resource `services` and verb `create` is new

File: addons/ekco/0.9.0/rbac.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    verbs:
+      - get
+      - list
+      - delete
+      - update
+      - create
+  - apiGroups: ["scheduling.k8s.io"]
```
Comment (areed) [inline addons/ekco/0.9.0/rbac.yaml:24]: this item and next are new

File: addons/ekco/0.9.0/install.sh (hunk 1 above)
Comment (laverya) [reply addons/ekco/0.9.0/install.sh:28]: is there a way for someone to turn off rook priority classes?

Comment (areed) [reply addons/ekco/0.9.0/install.sh:28]: You can manually edit the ekco configmap in the kurl namespace and remove the rook-priority-class option.

**Review by laverya — APPROVED** [review]

**Pull request merged**